grammar/helpers.js

```diff
@@ -0,0 +1,41 @@
+
+
+class Helpers {
+      /*
+    * @param {*} multiArr : The Array that newArr's values will be put into
+    * @param {*} inputArr : either an array of arrays, or an array with no sub arrays
+    * 
+    * Each sub array of either array represents a new line in a spreadsheet.
+    * updateRow appends all of the inputArr's values to the end of multiArr sequentially,
+    * such that the columns of multiArr and inputArr align
+    *  
+    * updateRow([[1], [3]], [[2], [3]])
+    * becomes:
+    * multiRow = [[1,2], [3,4]]
+    * on the spread sheet:
+    * 1 2
+    * 3 4
+    */
+    updateRow (multiArr, inputArr) {
+        if(!Array.isArray(multiArr))
+            throw "ERROR: multiArray is not array";
+        if (multiArr[0] === undefined || multiArr.length === 0)
+            multiArr.push([])
```

Comment:
The first condition here can be removed.

The only way the first condition here is true while the second condition is false is if  `multiArr` === `[undefined]` (could actually have any number of elements, but the first one is `undefined`). If that is the case, `multiArr` ends up being `[undefined, []]` after L23, and eventually `undefined.push()` will be called.

The way the code is written, `multiArr` always starts as `[[]]`, and we only push to it, so we'd have to use this function in a new place with bad input before these checks are relevant (still good to have them though!).

What we ultimately want, I think, is to confirm that every existing element of `multiArr[]` is an array (similar to check on L30 for elements of `inputArr`).  

Could either check up front:

```suggestion
        if (!multiArr.every(Array.isArray))
            throw "ERROR: Not all elements of multiArray are arrays";
        if (multiArr.length === 0)
            multiArr.push([]);        
```

(Note: `[].every(Array.isArray) === true`, since there is no element to check, so these `if`s work in either order ^^.)

Or check towards the end of the loop below:

```js
            if (!Array.isArray(multiArr[newArrIndex]))
                throw "ERROR: not all values are SubArrays";
        
            multiArr[newArrIndex].push(...currVal);
````